tools/test-proxy/Azure.Sdk.Tools.TestProxy/Sanitizers/BodyKeySanitizer.cs

```diff
@@ -25,7 +25,7 @@ public class BodyKeySanitizer : RecordedTestSanitizer
         /// <param name="value">The substitution value.</param>
         /// <param name="regex">A regex. Can be defined as a simple regex replace OR if groupForReplace is set, a subsitution operation. Defaults to replacing the entire string.</param>
         /// <param name="groupForReplace">The regex capture group that needs to be operated upon. Do not set if you're invoking a simple replacement operation.</param>
-        public BodyKeySanitizer(string jsonPath, string value = "Sanitized", string regex = ".*", string groupForReplace = null)
+        public BodyKeySanitizer(string jsonPath, string value = "Sanitized", string regex = ".+", string groupForReplace = null)
```

Comment:
Are you sure these defaults are being respected? While testing I found that I needed to pass the value explicitly.

Reply:
So, from what I saw, the defaulting is working how I expected. My unit test which loads the bodykeysanitizer explicitly does not pass a `regex` value.

I very possibly could be misunderstanding something though, so I'll take a look in the morning.

Reply:
Interesting, I'm not sure how it would be using the default value since we explicity pass null here: https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/Azure.Sdk.Tools.TestProxy/Admin.cs#L152

Reply:
Ok. With that change, I'm seeing the issue with `defaulting` that you mentioned. The piece I don't understand is why the regex is defaulting to `.+` successfully.

Taking a second look at Activator.CreateInstance. There's gotta be a method to make this do what I'm expecting.

Reply:
I think you can use parameter.DefaultValue instead of null.

Reply:
100%. That's the direction I'm going.